apps/web/playwright/booking-seats.e2e.ts

```diff
@@ -105,25 +113,108 @@ testBothBookers.describe("Booking with Seats", (bookerVariant) => {
-  // TODO: Make E2E test: Attendee #1 should be able to reschedule his booking
-  // todo("Attendee #1 should be able to reschedule his booking");
-  // TODO: Make E2E test: All attendees canceling should delete the booking for the User
-  // todo("All attendees canceling should delete the booking for the User");
+  // There are two URLs through which attendees can view the booking details to cancel, let's
+  // make sure both of them work
+  const seatedAttendeeUrls: [string, (bookingUid: string, seatUid: string) => string][] = [
+    [
+      "booking ref",
+      (bookingUid: string, seatUid: string) => `/booking/${bookingUid}?seatReferenceUid=${seatUid}`,
+    ],
+    ["seat ref", (bookingUid: string, seatUid: string) => `/booking/${seatUid}`],
+  ];
+  for (const [testRef, getBookingUrl] of seatedAttendeeUrls) {
+    test(`Attendees can cancel a seated event time slot (${testRef})`, async ({ page, users, bookings }) => {
+      // Cancelling via the "seat ref" does not behave correctly right now (it 404s after cancelling, and cancels the
+      // whole booking rather than just the relevant seat). We'll skip "seat ref" by default until fixed.
+      // eslint-disable-next-line playwright/no-skipped-test
+      test.skip(!process.env.E2E_TEST_SEATED_CANCELLATIONS && testRef == "seat ref");
+
+      const { booking } = await createUserWithSeatedEventAndAttendees({ users, bookings }, [
+        { name: "John First", email: "first+seats@cal.com", timeZone: "Europe/Berlin" },
+        { name: "Jane Second", email: "second+seats@cal.com", timeZone: "Europe/Berlin" },
+        { name: "John Third", email: "third+seats@cal.com", timeZone: "Europe/Berlin" },
+      ]);
+
+      const bookingAttendees = await prisma.attendee.findMany({
+        where: { bookingId: booking.id },
+        select: {
+          id: true,
+        },
+      });
+
+      const bookingSeats = [
+        { bookingId: booking.id, attendeeId: bookingAttendees[0].id, referenceUid: uuidv4() },
+        { bookingId: booking.id, attendeeId: bookingAttendees[1].id, referenceUid: uuidv4() },
+        { bookingId: booking.id, attendeeId: bookingAttendees[2].id, referenceUid: uuidv4() },
+      ];
+
+      await prisma.bookingSeat.createMany({
+        data: bookingSeats,
+      });
+
+      await test.step("Attendee #1 should be able to cancel their booking", async () => {
+        await page.goto(getBookingUrl(booking.uid, bookingSeats[0].referenceUid));
+
+        await page.locator('[data-testid="cancel"]').click();
+        await page.fill('[data-testid="cancel_reason"]', "Double booked!");
+        await page.locator('[data-testid="confirm_cancel"]').click();
+        await page.waitForLoadState("networkidle");
+
+        await expect(page).toHaveURL(/.*booking/);
+
+        const cancelledHeadline = await page.locator('[data-testid="cancelled-headline"]').innerText();
+        expect(cancelledHeadline).toBe("You are no longer attending this event");
```

Comment:
@zomars Can we rely on this message? or is it advisable to have another validation method? 

If this message changes the test will fail

Reply:
Agreed. We should expect the container to show. Something like `data-testid="error-message"`

Reply:
Good shout!

I ended up copying this `cancelledHeadline` pattern from existing tests:

https://github.com/calcom/cal.com/blob/974036239f204a21cc974026aa7366111b9012ca/apps/web/playwright/booking-pages.e2e.ts#L118-L120

Are we happy updating those existing usages alongside this PR, so we have consistency across tests?

Reply:
yes, just add it to the description for reference as you did with the others please.

Reply:
Done in 09b29ac03681d468db160233cedc296095afbb4c